packages/client/hmi-client/src/workflow/ops/stratify-mira/tera-stratify-mira.vue

```diff
@@ -126,41 +149,155 @@ const deleteStratifyGroupForm = (data) => {
 	});
 };
 
-const updateStratifyGroupForm = (data) => {
-	const state = _.cloneDeep(props.node.state);
-	state.strataGroups[data.index] = data.updatedConfig;
+const stratifyModel = () => {
+	stratifyRequest();
+};
 
-	workflowEventBus.emitNodeStateChange({
-		workflowId: props.node.workflowId,
-		nodeId: props.node.id,
-		state
+const resetModel = () => {
+	const kernel = jupyterSession.value?.session?.kernel;
+	if (!kernel || !model.value) return;
+
+	const messageBody = {
+		session: jupyterSession?.value?.name || '',
+		channel: 'shell',
+		content: {},
+		msgType: 'reset_request',
+		msgId: createMessageId('stratify_request')
+	};
+	const contextMessage: JupyterMessage = createMessage(messageBody);
+	kernel.sendJupyterMessage(contextMessage);
+};
+
+const stratifyRequest = () => {
+	const kernel = jupyterSession.value?.session?.kernel;
+	if (!kernel || !model.value) return;
+
+	const strataOption = props.node.state.strataGroups[0];
+	const messageBody = {
+		session: jupyterSession?.value?.name || '',
+		channel: 'shell',
+		content: {
+			stratify_args: {
+				key: strataOption.name,
+				strata: strataOption.groupLabels.split(',').map((d) => d.trim()),
+				concepts_to_stratify: strataOption.selectedVariables,
+				cartesian_control: strataOption.cartesianProduct
+			}
+		},
+		msgType: 'stratify_request',
+		msgId: createMessageId('stratify_request')
+	};
+	const contextMessage: JupyterMessage = createMessage(messageBody);
+	kernel.sendJupyterMessage(contextMessage);
+};
```

Comment:
As far as keeping history are we just going to throw the context message into strataCodeHistory here?

Reply:
Yes I think so - I will do that in the next PR.